unstructured/ingest/connector/clarifai.py

```diff
@@ -0,0 +1,128 @@
+import os
+import typing as t
+from dataclasses import dataclass, field
+import uuid
+from pathlib import PurePath
+
+from unstructured.ingest.enhanced_dataclass import enhanced_field
+from unstructured.ingest.enhanced_dataclass.core import _asdict
+from unstructured.ingest.interfaces import (
+    AccessConfig,
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseSingleIngestDoc,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.ingest.error import DestinationConnectionError
+from unstructured.ingest.utils.data_prep import chunk_generator
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
+if t.TYPE_CHECKING:
+    from clarifai.client.input import Inputs
+
+@dataclass
+class ClarifaiAccessConfig(AccessConfig):
+    api_key: str = enhanced_field(sensitive=True)
+    #print("class initialized")
+
+@dataclass
+class SimpleClarifaiConfig(BaseConnectorConfig):
+    access_config: ClarifaiAccessConfig
+    app_id: str
+    user_id: str
+    
+@dataclass
+class ClarifaiWriteConfig(WriteConfig):
+    batch_size: int = 50
+
+@dataclass
+class ClarifaiDestinationConnector(BaseDestinationConnector):
+    write_config: ClarifaiWriteConfig
+    connector_config: SimpleClarifaiConfig
+    _client: t.Optional["Inputs"] = field(init=False, default=None)
+    
+    @property
+    @requires_dependencies(["clarifai"], extras="clarifai")
+    def client(self) -> "Inputs":
+        if self._client is None:
+            from clarifai.client.input import Inputs
+        access_conf = self.connector_config.access_config
+        try:
+            if access_conf.api_key is not None :
+                clarifai_pat = access_conf.api_key   
+        except Exception as e:
+            raise (f"please provide clarifai PAT key : {e}")
+    
+        self._client = Inputs(app_id = self.connector_config.app_id, 
+                             user_id = self.connector_config.user_id, 
+                             pat = clarifai_pat)
+        return self._client
+        
+        
+    @requires_dependencies(["clarifai"], extras="clarifai")
+    @DestinationConnectionError.wrap
+    def initialize(self):
+        _ = self.client  
+       
+    def check_connection(self):
+        try:
+            _ = self.client
```

Comment:
Can we get something here that checks the live client on Clarifai? It's not enough to just create a client. (I do like that the _client can be created without a live connection. That is good.)

We usually try to do something like self._client.ping()

Reason being that we like to allow users to check that their connection input is correct without actually doing any uploading.

Reply:
Great, I have added list_inputs function(which yields a input generator obj) as a checkpoint, which works in a similar way to ping. It checks for the configuration and throws an error if it's not correct.